File: problem-solving/parsers/java/_05.java
import java.util.List;
import java.util.Scanner;

public class _05 implements Parser {
    public Object[] parser(Scanner f) {
        int n = f.nextInt();
        int m = f.nextInt();
        int maxTime = f.nextInt();
        List<Integer> energies = Parser.parse_list_int(f, n);
        List<List<Integer>> edges = Parser.parse_listlist_int(f, m, 3);

        assert(f.nextLine() == "");
        assert(f.nextLine() == "");

        int cert = f.nextInt();

        return Parser.ret_parser(cert, energies, edges, maxTime);
    }

    public Object run(Object[] input) {
        return Solutions.maximumMagicPathPower((List<Integer>)input[0], (List<List<Integer>>)input[1], (int)input[2]);
    }

    public boolean verifier(Object cert, Object ans) {
        return (int)cert == (int)ans;
    }

    public void error(Object cert, Object[] input, Object ans) {
        System.out.println(String.format(
            "Input:\n" +
            "energies = %s\n" +
            "edges = %s\n" +
            "maxTime = %d\n" +
            "\n" +
            "Expected:\n" +
            "%d\n" +
            "\n" +
            "Actual:\n" +
            "%d\n",
            Parser.print_list((List<Integer>)input[0]),
            Parser.print_listlist((List<List<Integer>>)input[1]),
            (int)input[2],
            (int)cert,
            (int)ans
        ));
    }

}

File: problem-solving/parsers/py/_05.py
def parser(f):
	n, m, maxTime = [ int(i) for i in f.readline().split() ]
	energies = [ int(f.readline()) for _ in range(n) ]
	edges = [ [int(i) for i in f.readline().split()] for _ in range(m) ]

	assert(f.readline() == "\n")

	cert = int(f.readline())
	assert(f.readline().strip() == "")

	return cert, (energies, edges, maxTime,)

def verifier(cert, ans):
	return cert == ans

def error(cert, input, ans):
	print("Input:")
	print("energies = " + str(input[0]))
	print("edges = " + str(input[1]))
	print("maxTime = " + str(input[2]))
	print("")
	print("Expected:")
	print(cert)
	print("")
	print("Actual:")
	print(ans)
	print()
